teuthology/suite/build_matrix.py

```diff
@@ -7,7 +7,7 @@
 log = logging.getLogger(__name__)
 
 
-def build_matrix(path, subset=None, seed=None):
+def build_matrix(path, subset=None, no_nested_subset=False, seed=None):
```

Comment:
Could you please add inline docs here about no_nested_subset?

Reply:
`nested_subset` should be the default so a negative option makes the most sense, to me.

Reply:
Thanks for adding the description. But I didn't understand why `nested_subset=True` does not work for you?

Reply:
Also, I wonder why haven't you put the new option at the end, so you do not need to change the code in hypothetic cases where build_matrix calls are not using it.

Reply:
> Thanks for adding the description. But I didn't understand why nested_subset=True does not work for you?

`nested_subset` should by default be `True`. Boolean options on the CLI are generally indicated by a simple switch, i.e. `--foo` without arguments. So, if we want it to be true by default, then the option should be `--no-foo`. This is not uncommon in UNIX, e.g. see `--no-seek` for GNU `tar` or `--no-clobber` in GNU `mv`.

Reply:
> Also, I wonder why haven't you put the new option at the end, so you do not need to change the code in hypothetic cases where build_matrix calls are not using it.

I fixed everywhere `build_matrix` is called. I felt it was sensible to group `subset` and `no_nested_subset` together in the argument list.

Reply:
> > Thanks for adding the description. But I didn't understand why nested_subset=True does not work for you?
> 
> `nested_subset` should by default be `True`. Boolean options on the CLI are generally indicated by a simple switch, i.e. `--foo` without arguments. So, if we want it to be true by default, then the option should be `--no-foo`. This is not uncommon in UNIX, e.g. see `--no-seek` for GNU `tar` or `--no-clobber` in GNU `mv`.

I understand your preferences for gnu switches, but the matter is to make function parameters, so no sense to use `no_` prefix everywhere, i.e. the switch can stay named `--no-nested-subset` but in function it is better to change to strict logic for nested subset.